Lib/test/test_zipfile/test_core.py

```diff
@@ -1080,6 +1080,127 @@ def test_generated_valid_zip64_extra(self):
                     self.assertEqual(zinfo.header_offset, expected_header_offset)
                     self.assertEqual(zf.read(zinfo), expected_content)
 
+    def test_force_zip64(self):
+        """Test that forcing zip64 extensions correctly notes this in the zip file"""
+        data = io.BytesIO()
+        with zipfile.ZipFile(data, mode="w", allowZip64=True) as zf:
+            with zf.open("text.txt", mode="w", force_zip64=True) as zi:
+                zi.write(b"_")
+
+        zipdata = data.getvalue()
+
+        # pull out and check zip information
+        (
+            header, vers, os, flags, comp, csize, usize, fn_len,
+            ex_total_len, filename, ex_id, ex_len, ex_usize, ex_csize, cd_sig
+        ) = struct.unpack("<4sBBHH8xIIHH8shhQQx4s", zipdata[:63])
+
+        self.assertEqual(header, b"PK\x03\x04")  # local file header
+        self.assertGreaterEqual(vers, zipfile.ZIP64_VERSION)  # requires zip64 to extract
```

Comment:
add comments describing where the #103861 bugs were before this test was added to the test.

Reply:
Added a comment to `test_force_zip64` mentioning the issue and specifically calling out the bits of data that were wrong. `test_unseekable_zip_known_filesize` also tests this issue but I didn't mention it in the comment on it directly since it's more about the zip64 interaction with the data descriptor than just checking if zip64 extensions are properly used.